zebra-consensus/src/block.rs

```diff
@@ -204,7 +202,12 @@ where
                 &block,
                 &transaction_hashes,
             ));
-            for transaction in &block.transactions {
+            for (n, transaction) in block.transactions.iter().enumerate() {
+                // Validate `nExpiryHeight` for non coinbase transactions
+                if n > 0 {
+                    check::non_coinbase_expiry_height(&height, transaction)?;
+                }
```

Comment:
Consensus rule: This rule also applies to the mempool.

Zebra checks if transactions in the mempool have expired. But we don't check expiry before adding transactions to the mempool.

Can you move the non-coinbase check to the transaction verifier, so both block and mempool transactions get checked? You can use `Transaction::has_valid_coinbase_transaction_inputs` to check for coinbase transactions.

Optional: You can also move the `coinbase_expiry_height` check if you want to keep them together. (The mempool rejects coinbase transactions, so it doesn't matter if the coinbase check is in the block or transaction verifier.)